soh/soh/Enhancements/randomizer/ShuffleCrates.cpp

```diff
@@ -0,0 +1,552 @@
+            default:
+                Gfx_DrawDListOpa(play, (Gfx*)gSmallJunkCrateDL);
+                break;
+        }
+}
+
+uint8_t ObjKibako2_RandomizerHoldsItem(ObjKibako2* crateActor, PlayState* play) {
+    RandomizerCheck rc = crateActor->crateIdentity.randomizerCheck;
+    uint8_t isDungeon = Rando::StaticData::GetLocation(rc)->IsDungeon();
+    uint8_t crateSetting = Rando::Context::GetInstance()->GetOption(RSK_SHUFFLE_CRATES).Get();
+
+    // Don't pull randomized item if crate isn't randomized or is already checked
+    if (!IS_RANDO || (crateSetting == RO_SHUFFLE_CRATES_OVERWORLD && isDungeon) ||
+        (crateSetting == RO_SHUFFLE_CRATES_DUNGEONS && !isDungeon) ||
+        Flags_GetRandomizerInf(crateActor->crateIdentity.randomizerInf) ||
+        crateActor->crateIdentity.randomizerCheck == RC_UNKNOWN_CHECK) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+uint8_t ObjKibako_RandomizerHoldsItem(ObjKibako* smallCrateActor, PlayState* play) {
+    RandomizerCheck rc = smallCrateActor->smallCrateIdentity.randomizerCheck;
+    uint8_t isDungeon = Rando::StaticData::GetLocation(rc)->IsDungeon();
+    uint8_t crateSetting = Rando::Context::GetInstance()->GetOption(RSK_SHUFFLE_CRATES).Get();
+
+    // Don't pull randomized item if crate isn't randomized or is already checked
+    if (!IS_RANDO || (crateSetting == RO_SHUFFLE_CRATES_OVERWORLD && isDungeon) ||
+        (crateSetting == RO_SHUFFLE_CRATES_DUNGEONS && !isDungeon) ||
+        Flags_GetRandomizerInf(smallCrateActor->smallCrateIdentity.randomizerInf) ||
+        smallCrateActor->smallCrateIdentity.randomizerCheck == RC_UNKNOWN_CHECK) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void ObjKibako2_RandomizerSpawnCollectible(ObjKibako2* crateActor, PlayState* play) {
+    EnItem00* item00 = (EnItem00*)Item_DropCollectible2(play, &crateActor->dyna.actor.world.pos, ITEM00_SOH_DUMMY);
+    item00->randoInf = crateActor->crateIdentity.randomizerInf;
+    item00->itemEntry = Rando::Context::GetInstance()->GetFinalGIEntry(crateActor->crateIdentity.randomizerCheck, true, GI_NONE);
+    item00->actor.draw = (ActorFunc)EnItem00_DrawRandomizedItem;
+    item00->actor.velocity.y = 8.0f;
+    item00->actor.speedXZ = 2.0f;
+    item00->actor.world.rot.y = Rand_CenteredFloat(65536.0f);
+}
+
+void ObjKibako_RandomizerSpawnCollectible(ObjKibako* smallCrateActor, PlayState* play) {
+    EnItem00* item00 = (EnItem00*)Item_DropCollectible2(play, &smallCrateActor->actor.world.pos, ITEM00_SOH_DUMMY);
+    item00->randoInf = smallCrateActor->smallCrateIdentity.randomizerInf;
+    item00->itemEntry =
+        Rando::Context::GetInstance()->GetFinalGIEntry(smallCrateActor->smallCrateIdentity.randomizerCheck, true, GI_NONE);
+    item00->actor.draw = (ActorFunc)EnItem00_DrawRandomizedItem;
+    item00->actor.velocity.y = 8.0f;
+    item00->actor.speedXZ = 2.0f;
+    item00->actor.world.rot.y = Rand_CenteredFloat(65536.0f);
+}
```

Comment:
Once we get the extendible actor structs these can probably be combined as the randomizerInf will probably live on actor level instead of `ObjKibako` level.